packages/domain/lib/validator/validator.dart

```diff
@@ -0,0 +1,54 @@
+abstract class Validator {
+  bool? isValid(Object a);
```

Comment:
don't make return type optional. throw error/return false instead of null. better to throw an error

Reply:
done with false. What are the advantages on throwing an error?

Reply:
for ex the new developer will try to validate int instead of string. and he didn't check the implementation of isValid. with exception it is easier to find out why validation is not correct